music21/roman.py

```diff
@@ -1111,6 +1214,28 @@ def romanNumeralFromChord(
         elif nationalityStart in ('Fr', 'Sw'):
             keyObj = _getKeyFromCache(chordObj.seventh.name.lower())
 
+    if (
+        preferSecondaryDominants
+        and stepRoman != 'V'  # ignore if already "primary" dominant
+        and (chordObj.isDominantSeventh() or chordObj.isMajorTriad())
+    ):
+        possibleSecondaryTonic = chordObj.root().transpose('P4').name
+        degree = keyObj.getScaleDegreeFromPitch(possibleSecondaryTonic)
+        if degree:  # None if not in chord
+            # Note: This is super verbose but the alternatives are bug-ridden. See #1450
+            fake2ndChordDegrees = [degree, degree + 2 % 8, degree + 4 % 8]
+            fake2ndChordPitches = keyObj.pitchesFromScaleDegrees(fake2ndChordDegrees)
+            fake2ndChord = chord.Chord(fake2ndChordPitches)
+            secondaryAsRoman = romanNumeralFromChord(fake2ndChord,
+                                                     keyObj,
+                                                     preferSecondaryDominants=False
+                                                     ).romanNumeralAlone
```

Comment:
Thanks for hacking something up. I'm just wondering, if we merge #1481 soon, which I think can happen, can this just use `RomanNumeral()` rather than `romanNumeralFromChord()`? That way, we neither have to wait for #1480, nor add this workaround.

Reply:
btw #1481 was merged.

Reply:
Yup, that now works. Thanks for the swift work.